### PR TITLE
Feature checkbox

### PR DESCRIPTION
**Pull request by metruzanca**
I think we need to make a template for PRs

(this is a test of how vercel handles PRs)

### CODE REVIEW

**Comment by zombieFox** [comment]
I've added the CSS needed to get the checkbox (and maybe the radio) working. Want me to merger it to this branch? 

Added files (which can be refactored later to be more modular):
src/styles/base.css
src/styles/form.css
src/styles/reset.css
src/styles/variables.css

Also changed /src/components/Checkbox/Checkbox.tsx to add a `form-wrap` div needed for all form elements. This wrapper (along with `form-indent` and `form-inline`) gives us the spacing and form structure.

**Comment by metruzanca** [comment]
Ohh, cool! I noticed while looking at v2 that theres actually 2 checkboxes, one thats wrapped in the form button (to make it a button) and the non-button checkbox. Are there many more elements that are "buttonized"? if so, we could consider making a form Higher order component. I'm not sure if you've heard these mentioned while learning react yet, but they are just like higher order functions, so they are functions that take a component and return a component, usually with either some kind of wrapper or some props injected into it or both.

However something we definetly need to do is make a form component which, correct me if I'm wrong, theres just the form and inline-form version and both are essentially a different type of grouping different from group.

**Comment by metruzanca** [comment]
> Want me to merge it to this branch?

yeah, go for it.

**Comment by zombieFox** [comment]
Yes, the radio, checkboxe, input file  input colour have "buttonised" versions. Buttons just felt right for some actions. I think that is all of them. We definitely want to abstract that out.

Yup, I think a "form" component which takes other form elements (like checkboxes) is a very good idea. I did that [on the hex project](https://github.com/zombieFox/hexagonTab/blob/ee9166566b71f48d5ce59a8a503b9e171c110c0f/src/js/utilities/form.js#L8). Those functions just wrap a given element with the right `form-*` tag. There are five types: 

- fieldset (wrap for multiple form elements)
- form-wrap (wrap for individual form elemet)
- form-sticky (to stick a given form-wrap on scroll)
- form-group (take given form inputs and make them look like a single horizontal block)
- form-indent (spacing on the left of a given form area)

> yeah, go for it.

Cool!

File: src/index.tsx [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
+import './styles/reset.css';
+import './styles/base.css';
+import './styles/form.css';
+import './styles/variables.css';
```
Comment (metruzanca) [inline src/index.tsx:7]:
```suggestion
import 'styles/reset.css';
import 'styles/base.css';
import 'styles/form.css';
import 'styles/variables.css';
```
Doesn't really matter here as we're in the root, but just as a rule of thumb try to favor absolute paths as it removes unecessary noise.

Comment (zombieFox) [reply src/index.tsx]: Can do 👍

Comment (metruzanca) [reply src/index.tsx]: FYI, github has a "commit suggestion" button.

Comment (zombieFox) [reply src/index.tsx]: Yup, that's what I used 👍

**Comment by metruzanca** [comment]
Feel free to merge this branch, we'll likely re-iterate on checkbox once we've got the form worked out. Then feel free to fix the button's styles too and work on other components if you like.

I'll work on getting the state management going. Leaning towards contexts as I don't know if redux would add a noticeable delay on startup with big configs. I'll see to get that done on wednesday.

Cheers

**Pull request merged**